plugins/vst_base/RemoteVstPlugin.cpp

```diff
@@ -577,27 +646,39 @@ bool RemoteVstPlugin::processMessage( const message & _m )
 			return true;
 
 		case IdToggleUI:
+#ifndef NATIVE_LINUX_VST
 			if( m_window && IsWindowVisible( m_window ) )
+#else
+			if (m_window && m_x11WindowVisible)
+#endif
 			{
 				hideEditor();
 			}
 			else
 			{
 				showEditor();
 			}
+
 			return true;
 
 		case IdIsUIVisible:
+			debugMessage("is visible?");
```

Comment:
Don't forget to remove some unnecessary `debugMessage` calls once you're done.

Reply:
Removed